numpy/f2py/crackfortran.py

```diff
@@ -2873,6 +2909,141 @@ def analyzevars(block):
 analyzeargs_re_1 = re.compile(r'\A[a-z]+[\w$]*\Z', re.I)
 
 
+def param_eval(n, varsn, v, g_params, params):
+    """
+    Creates a dictionary of indices and values for each parameter in a
+    parameter array to be evaluated later.
+
+    WARNING: It is not possible to initialize multidimensional array
+    parameters e.g. dimension(-3:1, 4, 3:5) at this point. This is because in
+    Fortran initialization through array constructor requires the RESHAPE
+    intrinsic function. Since the right-hand side of the parameter declaration
+    is not executed in f2py, but rather at the compiled c/fortran extension,
+    later, it is not possible to execute a reshape of a parameter array.
+    One issue remains: if the user wants to access the array parameter from
+    python, we should either
+    1) allow them to access the parameter array using python standard indexing
+       (which is often incompatible with the original fortran indexing)
+    2) allow the parameter array to be accessed in python as a dictionary with
+       fortran indices as keys
+    We are choosing 2 for now.
+    """
+    dim_attrspecs = [item[9:] for item in varsn['attrspec']
+                     if item[:9] == 'dimension']
+    if len(dim_attrspecs) > 1:
+        raise ValueError('dimension may appear at most once.')
+    elif dim_attrspecs:
+        # This is an array parameter.
+        # First, we parse the dimension information
+        a = dim_attrspecs[0]
+        a = a.strip()
+        if a[0] != "(" or a[-1] != ")":
+            raise ValueError(f'dimension for {n} can\'t be parsed')
+        dimrange = a[1:-1]
+        dimrange = dimrange.split(',')
+        if len(dimrange) == 1:
+            # e.g. dimension(2) or dimension(-1:1)
+            dimrange = dimrange[0].split(':')
+            # now, dimrange is a list of 1 or 2 elements
+            if len(dimrange) == 1:
+                bound = param_parse(dimrange[0], params)
+                dimrange = range(1, int(bound)+1)
+            else:
+                lbound = param_parse(dimrange[0], params)
+                ubound = param_parse(dimrange[1], params)
+                dimrange = range(int(lbound), int(ubound)+1)
+        else:
+            outmess('param_eval: multidimensional array parameters not'
+                    ' supported: %s\n' % repr(n))
+
+        # Parse parameter value
+        v = v.lstrip('(/').rstrip('/)').split(',')
```

Comment:
strip is likely not suitable here, it maps `(/(a), b/)` to `a), b`.

Reply:
I updated the code to address this, but I'm not sure if another approach would be preferable.

Reply:
```
v = (v[2:-2] if v.startswith('(/') else v).split(',')
```